### PR TITLE
Add w3cDate modelType for date picker.

### PR DESCRIPTION
**Pull request by mattcollier**

### CODE REVIEW

File: datepicker-directive.js [hunk 1]
```diff
@@ -125,6 +125,8 @@ function datepickerFactory($filter, $templateCache, $timeout) {
         }
         if(ctrl.options.modelType === 'string') {
           ctrl.model = dateFilter(ctrl.date, ctrl.options.format);
+        } if(ctrl.options.modelType === 'w3cDate') {
+          ctrl.model = util.w3cDate(ctrl.date);
```
Comment (dlongley) [inline datepicker-directive.js]: Doesn't using `modelType = 'string'` already do the same thing? You just need to set the appropriate format? If not, maybe we just need a `modelFormat` option? Adding `w3cDate` seems really specific.

**Comment by mattcollier** [comment]
@dlongley another pass.

File: datepicker-directive.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ define([], function() {
 var templateReplaced = false;
 
 /* @ngInject */
-function datepickerFactory($filter, $templateCache, $timeout) {
+function datepickerFactory($filter, $templateCache, $timeout, util) {
```
Comment (dlongley) [inline datepicker-directive.js]: `util` now not needed, right?

**Comment by dlongley** [comment]
One comment to address, then LGTM, merge.

**Pull request merged**